v2/check_logger.go

```diff
@@ -0,0 +1,9 @@
+package voucher
+
+type CheckLogger interface {
+	LogDebug(check, callerFile, callerFunction, msg string)
```

Comment:
Manually specifying `check`, `callerFile`, and `callerFunction` smells a little strange to me.
You could simplify this interface by using `runtime.Caller()` to look these values up - https://go.dev/play/p/VRKdukRDFGw

If you keep pulling at this thread, `logrus.FieldLogger` could be an alternative to the custom `CheckLogger` interface.